ipam/client/backends/phpipam.py

```diff
@@ -218,6 +218,54 @@ def add_top_level_subnet(self, subnet, description):
 
         return True
 
+    def add_subnet(self, subnet, parent_subnet, description):
+        """
+        Add a subnet if can be inserted in parent subnet.
+        """
+        parent_subnet_id = self.find_subnet_id(parent_subnet)
+
+        subnet_len = subnet.prefixlen
+        if not subnet.overlaps(parent_subnet):
+            raise ValueError('Subnet {} is not a child of {}'.format(
+                subnet,
+                parent_subnet,
+            ))
+
+        children_subnets = self.get_children_subnet_list(parent_subnet)
+        for children_subnet in children_subnets:
+            if children_subnet['subnet'].overlaps(subnet):
+                raise ValueError('Candidate subnet overlaps with {}'.format(
+                    children_subnet['description']
+                ))
+
+        parent_subnet_used_ips = self.get_allocated_ips_by_subnet_id(
+            parent_subnet_id)
+        if len(parent_subnet_used_ips) > 0:
+            raise ValueError('Parent subnet {} must not contain any '
+                             'allocated IP address!'.format(parent_subnet))
+
+        # check if parent_subnet has enough
+        # space to add a subnet_len sized subbnet
+        _ = self._get_next_free_subnet(parent_subnet, parent_subnet_id,
```

Comment:
Do you really need this? If subnet overlaps parent one, then it fits in it, and if it doesn't overlap children, then you're sure it doesn't conflict with them.